### PR TITLE
add hugging face sentiment analysis api

### PR DESCRIPTION
**Pull request by khanayan123**
add hugging face sentiment analysis api

### CODE REVIEW

**Comment by Wayakuyee** [comment]
What do we need to install to run the code? I installed transformers and tried to run it but now it tells me I need PyTorch, TensorFlow, or Flax so I wanna ask in case there is more to install that just those two.

image

**Comment by khanayan123** [comment]
> What do we need to install to run the code? I installed transformers and tried to run it but now it tells me I need PyTorch, TensorFlow, or Flax so I wanna ask in case there is more to install that just those two.
> 
> image

right i will update that

File: parsing_and_analysis/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+wurlitzer @ file:///opt/concourse/worker/volumes/live/a2b1dfd8-5830-41c1-6a08-9a84d13fec28/volume/wurlitzer_1638368180115/work
+xarray @ file:///opt/conda/conda-bld/xarray_1639166117697/work
+xlrd @ file:///tmp/build/80754af9/xlrd_1608072521494/work
+XlsxWriter @ file:///opt/conda/conda-bld/xlsxwriter_1649073856329/work
+xlwings==0.24.9
+yapf @ file:///tmp/build/80754af9/yapf_1615749224965/work
+yarl @ file:///opt/concourse/worker/volumes/live/8e0192c5-7b7d-47fe-5cfd-fadef81d5ebb/volume/yarl_1606940005549/work
+zict==2.0.0
+zipp @ file:///opt/conda/conda-bld/zipp_1641824620731/work
+zope.interface @ file:///opt/concourse/worker/volumes/live/8c2d4bd1-6406-47f0-45ce-890992bafbf6/volume/zope.interface_1625036159007/work
```
Comment (alvii147) [inline parsing_and_analysis/requirements.txt:344]: There are way too many dependencies here, I think you may have submitted all the dependencies currently in your system. I doubt we need all of these, a lot of them have nothing to do with our codebase. Can you recheck this? Try using a [virtual environment](https://docs.python.org/3/library/venv.html) for this.

Also, some of these entries are a little weird, for e.g.:

```
boto3 @ file:///opt/conda/conda-bld/boto3_1649078879353/work
```

That doesn't look right. Take a look at this [stack overflow post](https://stackoverflow.com/questions/62885911/pip-freeze-creates-some-weird-path-instead-of-the-package-version).

File: parsing_and_analysis/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+The requirements is a bit overloaded, but heres what you need to install to get it to run:
+
+*Ensure you are using python 3.7+
+
+1. pip3 install transformers
+2. pip3 install tensorflow
+
```
Comment (alvii147) [inline parsing_and_analysis/README.md:7]: We shouldn't be installing like this without specifying a version. The requirements should be installed by doing `pip3 install -r requirements.txt`.

File: parsing_and_analysis/main.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+emotion = pipeline('sentiment-analysis', 
+                    model='arpanghoshal/EmoRoBERTa')
+
+
+def analyzeText(input):
+    emotion_labels = emotion(input)
+    return emotion_labels
+
+
```
Comment (alvii147) [inline parsing_and_analysis/main.py:15]: This looks good, can you add some comments that explain how to use this? Possibly mirroring the comments or docstrings in the rest of the Python modules.

File: parsing_and_analysis/main.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from transformers import RobertaTokenizerFast, TFRobertaForSequenceClassification, pipeline
+
+tokenizer = RobertaTokenizerFast.from_pretrained("arpanghoshal/EmoRoBERTa")
+model = TFRobertaForSequenceClassification.from_pretrained("arpanghoshal/EmoRoBERTa")
+
+emotion = pipeline('sentiment-analysis', 
+                    model='arpanghoshal/EmoRoBERTa')
+
+
+def analyzeText(input):
```
Comment (alvii147) [inline parsing_and_analysis/main.py:11]: Might wanna change the variable name from `input` to something else as `input` is a Python special keyword. Maybe `input_text`?

**Review by alvii147 — CHANGES_REQUESTED** [review]

**Comment by alvii147** [comment]
This looks good. The only thing I'll say is maybe you can take this file out of the `parsing_and_analysis` directory, and just put it in the `backend/` directory because that's where most of the backend stuff is. I know the backend directory is getting crowded, but we'll tidy things up there soon.

**Review by Wayakuyee — APPROVED** [review]
Removed the unnecessary parts in the README, LGTM!

**Review by alvii147 — APPROVED** [review]

**Pull request merged**